pkg/ext/resources/tokens/store.go

```diff
@@ -0,0 +1,325 @@
+		authed, err := t.userHasFullPermissions(userInfo)
+		if err != nil {
+			return nil, fmt.Errorf("unable to check if user has * on tokens: %w", err)
+		}
+		if !authed {
+			return nil, fmt.Errorf("can't create token for other user %s since user %s doesn't have * on ranchertokens", userInfo.GetName(), token.Spec.UserID)
+		}
+	}
+	token.Status.PlaintextToken = ""
+	return token, nil
+}
+
+func (t *TokenStore) List(ctx context.Context, userInfo user.Info, opts *metav1.ListOptions) (*RancherTokenList, error) {
+	secrets, err := t.secretClient.List(TokenNamespace, *opts)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list tokens: %w", err)
+	}
+	authed, err := t.userHasFullPermissions(userInfo)
+	if err != nil {
+		return nil, fmt.Errorf("unable to check if user has * on tokens: %w", err)
+	}
+	var tokens []RancherToken
+	for _, secret := range secrets.Items {
+		token := tokenFromSecret(&secret)
+		// users can only list their own tokens, unless they have * on this group
+		if !authed && token.Spec.UserID != userInfo.GetName() {
+			continue
+		}
+		tokens = append(tokens, *token)
+	}
+	list := RancherTokenList{
+		ListMeta: metav1.ListMeta{
+			ResourceVersion: secrets.ResourceVersion,
+		},
+		Items: tokens,
+	}
+	return &list, nil
+}
+
+// TODO: Close channel
+func (t *TokenStore) Watch(ctx context.Context, userInfo user.Info, opts *metav1.ListOptions) (<-chan types.WatchEvent[*RancherToken], error) {
+	authed, err := t.userHasFullPermissions(userInfo)
+	if err != nil {
+		return nil, fmt.Errorf("unable to check if user has * on tokens: %w", err)
+	}
+
+	ch := make(chan types.WatchEvent[*RancherToken])
+
+	go func() {
+		watcher, err := t.secretClient.Watch(TokenNamespace, metav1.ListOptions{
+			ResourceVersion: opts.ResourceVersion,
+		})
+		if err != nil {
+			return
+		}
+		defer watcher.Stop()
+
+		for event := range watcher.ResultChan() {
+			secret, ok := event.Object.(*corev1.Secret)
+			if !ok {
+				continue
+			}
+
+			token := tokenFromSecret(secret)
+			if !authed && token.Spec.UserID != userInfo.GetName() {
+				continue
+			}
+
+			watchEvent := types.WatchEvent[*RancherToken]{
+				Event:  event.Type,
+				Object: token,
+			}
+			ch <- watchEvent
+		}
+	}()
+
+	return ch, nil
+}
+
+func (t *TokenStore) Delete(ctx context.Context, userInfo user.Info, name string, opts *metav1.DeleteOptions) error {
+	fmt.Println(userInfo, name)
+	secret, err := t.secretCache.Get(TokenNamespace, name)
+	if err != nil {
+		return fmt.Errorf("unable to confirm secret existence %s: %w", name, err)
+	}
+	token := tokenFromSecret(secret)
+	// user is trying to get a token for another user
+	if token.Spec.UserID != userInfo.GetName() {
+		authed, err := t.userHasFullPermissions(userInfo)
+		if err != nil {
+			return fmt.Errorf("unable to check if user has * on tokens: %w", err)
+		}
+		if !authed {
+			return fmt.Errorf("can't create token for other user %s since user %s doesn't have * on ranchertokens", userInfo.GetName(), token.Spec.UserID)
```

Comment:
```suggestion
			return fmt.Errorf("can't delete token for other user %s since user %s doesn't have * on ranchertokens", userInfo.GetName(), token.Spec.UserID)
```